backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,319 @@
+package backtraceio.library.breadcrumbs;
+
+import android.app.Application;
+import android.content.Context;
+
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+import backtraceio.library.enums.BacktraceBreadcrumbLevel;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+import backtraceio.library.interfaces.Breadcrumbs;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.models.json.BacktraceReport;
+
+public class BacktraceBreadcrumbs implements Breadcrumbs {
+
+    private static transient String LOG_TAG = BacktraceBreadcrumbs.class.getSimpleName();
+
+    /**
+     * Which breadcrumb types are enabled?
+     */
+    private EnumSet<BacktraceBreadcrumbType> enabledBreadcrumbTypes;
+
+    /**
+     * The Backtrace BroadcastReciever instance
+     */
+    private BacktraceBroadcastReceiver backtraceBroadcastReceiver;
+
+    /**
+     * The Backtrace ComponentCallbacks2 listener
+     */
+    private BacktraceComponentListener backtraceComponentListener;
+
+    /**
+     * The Backtrace ActivityLifecycleCallbacks listener
+     */
+    private BacktraceActivityLifecycleListener backtraceActivityLifecycleListener;
+
+    /**
+     * The Backtrace Breadcrumbs log manager
+     */
+    private BacktraceBreadcrumbsLogManager backtraceBreadcrumbsLogManager;
+
+    private Context context;
+
+    private static final int DEFAULT_MAX_LOG_SIZE_BYTES = 64000;
+
+    String breadcrumbLogDirectory;
+
+    final private static String breadcrumbLogFileName = "bt-breadcrumbs-0";
+
+    public BacktraceBreadcrumbs(String breadcrumbLogDirectory) {
+        this.breadcrumbLogDirectory = breadcrumbLogDirectory;
+    }
+
+    private void registerAutomaticBreadcrumbReceivers() {
+        if (enabledBreadcrumbTypes == null) {
+            return;
```

Comment:
I think that it's worth to log information about it

Reply:
Done, thank you!